.circleci/config.yml

```diff
@@ -10,35 +10,62 @@ jobs:
     steps:
       - checkout
 
-      #TODO: add caching once demos share the same requirements file and can live in the same venv
-      #TODO: remove hard code in for-loops
+      # Restore cache
+      - run: &set-bash-env-template
+          name: Set bash environment
+          command: |
+            # Storing values shared between run calls
+            # Note: TEMP_FILE value is hardcoded in restore_cache and save_cache's checksum
+            echo 'export TEMP_FILE="temp_requirements_summary.txt"' >> $BASH_ENV
```

Comment:
Is this guaranteed to work on CircleCI?

Because, generally, `BASH_ENV` is **not** defined. In which case this whole block of commands will fail.

Reply:
I learned this trick from the CircleCI docs: https://circleci.com/docs/2.0/env-vars/#example-configuration-of-environment-variables

Based on the fact that it's how they're teaching people to share variables between runs, I am assuming it's guaranteed to work on CircleCI

Reply:
Using job environment variable now!

Reply:
Ah, here's the [explanation](https://circleci.com/docs/2.0/env-vars/#using-bash_env-to-set-environment-variables) why they use that horrible hack -- job environment variables support only fixed/hardcoded values. But that's good enough here.